### DISCUSSION TITLE
Metrics before `An error occured: Runner version v2.310.2 is deprecated and cannot receive messages.`

### DISCUSSION THREAD

**Discussion by MPV** [community/community#160685] [Actions]
Hi,
Could ARC please get a metric for showing which runner scale sets have outdated versions (and have started getting deprecation warnings)?

Ideally one doesn’t want to run into the (too late) error:
`An error occured: Runner version v2.310.2 is deprecated and cannot receive messages.`

And rather get it as early as when you start getting job annotations such as:
`This job failure may be caused by using an out of date self-hosted runner. You are currently using runner version 2.310.2. Please update to the latest version 2.311.0`

See here for another example/motivation around the same problem:
- https://github.com/actions/actions-runner-controller/issues/2943#issuecomment-1770417330

For reference:
- https://github.com/actions/runner/issues/1991
- https://github.com/actions/runner/issues/1843#issuecomment-1183295528

**Comment by wad-hongsumin** [comment]
The version of the runner in the image has been deprecated. In my case, I was using my own Dockerfile based on the ubuntu-20.04 version in October. I found the same error as you and solved it by building a new Dockerfile and updating the image.

The issue I referenced:
https://github.com/actions/runner/issues/1991

**Reply by MPV** [reply]
I'm aware this is the problem.

But I would like to measure it before/when it becomes a problem, so I can alert on it, and thus act on it before developers using the runners experience the issue and reach out about the problem. 

Hence my suggestion about exposing a metric about this.

**Comment by MPV** [comment]
Could the API mentioned here be an option to use for understanding when runners are of an outdated version?

- https://github.com/actions/runner/issues/1843#issuecomment-1503056202 👇
   - https://docs.github.com/en/github-ae@latest/rest/actions/self-hosted-runners#list-runner-applications-for-an-organization

**Comment by arslan-autoscout24** [comment]
This is really important for us as well. All the pipelines broken suddenly is not a good condition. I would really appreciate if there is a workaround to avoid this situation.
